Adafruit_DAP_SAM.cpp

```diff
@@ -128,13 +130,101 @@ bool Adafruit_DAP_SAM::select(uint32_t *found_id) {
     if (device->dsu_did == DAP_DSU_did) {
       target_device = *device;
 
+      locked = dap_read_word(DAP_DSU_CTRL_STATUS) & 0x00010000;
+      if (locked) {
+        Serial.println("Device is locked, must be unlocked!");
+      }
+      else {
+        // Stop the core
+        finishReset();
+      }
       return true;
     }
   }
 
   return false;
 }
 
+void Adafruit_DAP_SAM::finishReset() {
+	// Stop the core
+	dap_write_word(DHCSR, 0xa05f0003);
+	dap_write_word(DEMCR, 0x00000001);
+	dap_write_word(AIRCR, 0x05fa0004);
```

Comment:
do we need to release the reset as edbg did

https://github.com/ataradov/edbg/blob/master/target_atmel_cm0p.c#L173

Reply:
Maybe not, but it was released in the original code. Didn't want to change the functionality.

Reply:
no problem at all, I could take a look closer at this later on with hand-on testing on existing targets to verify.